packages/react-aria-components/src/Table.tsx

```diff
@@ -312,7 +313,18 @@ export interface TableProps extends Omit<SharedTableProps<any>, 'children'>, Sty
   /** Handler that is called when a user performs an action on the row. */
   onRowAction?: (key: Key) => void,
   /** The drag and drop hooks returned by `useDragAndDrop` used to enable drag and drop behavior for the Table. */
-  dragAndDropHooks?: DragAndDropHooks
+  dragAndDropHooks?: DragAndDropHooks,
+  // TODO: finalize these
+  // Perhaps we should make isLoading be loadingKeys for the eventual section/nested rows loading case
+  // Or should it follow the loadingState api we have for our async hook
+  // TODO: should the below be loadMoreOptions? These are just passed to the hook and making a prop that takes in a object containing the
+  // options might be easier to adjust in the future
+  isLoading?: boolean,
+  onLoadMore?: () => void,
+  scrollOffset?: number,
+  // TODO: keep this for now since we might need it for the body, but eventually perhaps we could just do tableContainerContext?.scrollRef || table ref if it turns out we
+  // don't need the body to be scrollable
+  scrollRef?: RefObject<HTMLElement>
```

Comment:
Some api question for discussion

Reply:
When we finalize the api/useLoadOnScroll, I'll go ahead and add this to all the other collection components